Automattic/wp-calypso
# Framework: Introduce top-level Redux state directory with sites example

__Update:__ This pull request was originally targeted toward re-working `sites-list`, but it has since grown to include changes for adopting a new folder structure approach with regards to our Redux global state. Specifically, all files related to the global state tree are now contained in a new top-level `client/state` directory. The directory reflects the structure of the global state object itself, e.g. `state.sites` in the global state maps to files contained in `client/state/sites`.

Many other guidelines are yet to be developed. See #1378 for progress in crafting a recommendations document.

----

Blocks #756

This pull request seeks to initiate the effort to refactor the existing `sites-list` module to reflect our current data management recommendations. The changes herein are minimal in that the only features ported are actions to set a selected site and to receive a site. A follow-up pull request intends to dispatch these actions in the [My Sites `siteSelection` controller middleware function](https://github.com/Automattic/wp-calypso/blob/f62b9366661fc291336d98cdcf600e9d789964fa/client/my-sites/controller.js#L81-L152), thus making the selected site available from within global Redux state.

Refer to the [module README.md](https://github.com/Automattic/wp-calypso/blob/add/sites-redux/shared/lib/sites/README.md) for more information.

__Testing instructions:__

There are no user flows which will cause the added modules to be imported. Therefore, there are no effective changes in the application. Verify instead that the included tests pass by running `make test` from the `shared/lib/sites` directory.
